src/effects/effectchainslot.cpp

```diff
@@ -245,19 +245,11 @@ EffectChainPointer EffectChainSlot::getEffectChain() const {
 }
 
 void EffectChainSlot::clear() {
-    // Stop listening to signals from any loaded effect
     if (m_pEffectChain) {
-        m_pEffectChain->removeFromEngine(m_pEffectRack->getEngineEffectRack(),
-                                         m_iChainSlotNumber);
-        foreach (EffectSlotPointer pSlot, m_slots) {
+        for (EffectSlotPointer pSlot: m_slots) {
             pSlot->clear();
         }
-        m_pEffectChain->disconnect(this);
-        m_pEffectChain.clear();
```

Comment:
I think the concept of loading and ejecting chains from the engine is useless now. When we implement saving and loading effect chain presets, I think it will be implemented as loading the preset state with the effects and chains already loaded in the engine.

Checking for a null pointer where the crash happens would only be a graceful exit, not actually fix the issue. If the pointer was null, then no effects could be loaded in the chain.